File: banchus-common/src/main/java/pe/nanamochi/banchus/database/repository/BeatmapsetRepository.java
package pe.nanamochi.banchus.database.repository;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import pe.nanamochi.banchus.database.entity.Beatmapset;

@Repository
public interface BeatmapsetRepository extends JpaRepository<Beatmapset, Integer> {
  Optional<Beatmapset> findById(int id);
}

File: banchus-common/src/main/kotlin/pe/nanamochi/banchus/database/repository/BeatmapsetRepository.kt
package pe.nanamochi.banchus.database.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import pe.nanamochi.banchus.database.entity.Beatmapset

@Repository
interface BeatmapsetRepository : JpaRepository<Beatmapset, Int> {
    fun findBeatmapsetById(id: Int): Beatmapset?
}
